File: app/src/main/java/com/antyzero/smoksmog/ui/screen/FragmentComponent.java
package com.antyzero.smoksmog.ui.screen;

import com.antyzero.smoksmog.ui.screen.settings.GeneralSettingsFragment;
import com.antyzero.smoksmog.ui.screen.start.fragment.NetworkStationFragment;

import dagger.Subcomponent;

@Subcomponent(
        modules = {
                FragmentModule.class,
        }
)
public interface FragmentComponent {

    void inject(GeneralSettingsFragment generalSettingsFragment);

    void inject(NetworkStationFragment networkStationFragment);
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/FragmentComponent.kt
package com.antyzero.smoksmog.ui.screen

import com.antyzero.smoksmog.ui.screen.settings.GeneralSettingsFragment
import com.antyzero.smoksmog.ui.screen.start.fragment.NetworkStationFragment

import dagger.Subcomponent

@Subcomponent(modules = arrayOf(FragmentModule::class))
interface FragmentComponent {

    fun inject(generalSettingsFragment: GeneralSettingsFragment)

    fun inject(networkStationFragment: NetworkStationFragment)
}
